src/file-selector.ts

```diff
@@ -108,34 +108,51 @@ function flatten<T>(items: any[]): T[] {
 async function fromDataTransferItem(
   item: DataTransferItem,
   entry?: FileSystemEntry | null,
-) {
-  // Check if we're in a secure context; due to a bug in Chrome (as far as we know) the browser crashes when calling this API (yet to be confirmed as a consistent behavior).
-  //
+): Promise<FileWithPath> {
+  const fileWithPath = await fromFileSystemHandle(item, entry);
+
+  if (fileWithPath) {
+    return fileWithPath;
+  }
+
+  const file = item.getAsFile();
+
+  if (!file) {
+    throw new Error("Transferred item is not a file.");
+  }
+
+  return toFileWithPath(file, entry?.fullPath);
+}
+
+async function fromFileSystemHandle(
+  item: DataTransferItem,
+  entry?: FileSystemEntry | null,
+): Promise<File | null> {
+  // Check if we're in a secure context; otherwise due to a bug in Chrome the browser can crash when calling this API.
   // See:
   // - https://issues.chromium.org/issues/40186242
   // - https://github.com/react-dropzone/react-dropzone/issues/1397
+  // If the browser does not support the API act as if the file could not be retrieved.
   if (
-    globalThis.isSecureContext &&
-    typeof (item as any).getAsFileSystemHandle === "function"
+    !globalThis.isSecureContext ||
+    !(typeof item.getAsFileSystemHandle === "function")
   ) {
-    const h = await (item as any).getAsFileSystemHandle();
-    if (h === null) {
-      throw new Error(`${item} is not a File`);
-    }
-    // It seems that the handle can be `undefined` (see https://github.com/react-dropzone/file-selector/issues/120),
-    // so we check if it isn't; if it is, the code path continues to the next API (`getAsFile`).
-    if (h !== undefined) {
-      const file = await h.getFile();
-      file.handle = h;
-      return toFileWithPath(file);
-    }
+    return null;
```

Comment:
Since where this function is used, `null` is not explicitly checked, a simple:
```suggestion
    return;
```

would also work since `if (undefined)` is the same as `if (null)`. 

You can probably tell that I'm not a fan of adding unnecessary code when functionally a shorter version does the same (and there's really no gains in improving readability either).

Reply:
In this case I just follow what the API returns here as well, I don't like adding polymorphism where it is not needed. In most cases `null` is meant as a 'this operation yields an empty result' and `undefined` means 'this is void' (no value is ever returned). You'll see this convention in most APIs that exist in the standard API of JavaScript runtimes.